eng/common/scripts/Set-DefaultBranchVariable.ps1

```diff
@@ -0,0 +1,36 @@
+param (
+    [Parameter(Mandatory = $true)]
+    [string]$RepoOwner,
+
+    [Parameter(Mandatory = $true)]
+    [string]$RepoName
+)
+
+. (Join-Path $PSScriptRoot common.ps1)
+
+LogDebug "`$RepoOwner was: $RepoOwner"
+LogDebug "`$RepoName was: $RepoName"
+
+try
+{
+    if ($RepoName -like "*-pr") {
```

Comment:
While this might work in the general case there may still be other private repos that we need to support. Is there not a way to use the ACCESS_TOKEN in the pipeline to query github?

Reply:
Let me double check that, now that the ```- checkout:``` task can be moved around the credential must be hanging around somewhere.

Reply:
So my findings so far are that in order to do this we would need to create an Azure Pipelines task, which means that we would need to build an Azure DevOps extension. Given our future direction is probably GitHub Actions (at some point in the yet to be determined future) I feel like this would be a lot of effort for not too much gain.

Reply:
I agree if there isn't a easy way to do then we can have some hacks like your naming match. However I will say that it will mean that we need to do the rename in the pr repos at the exact same time so that we don't end up with a master and main in the pr repos. I will also say that the pr repos should be a lot easier to move because no one is every merging into master directly.

Reply:
Actually, during some reading I found this option:

```yaml
- checkout:
  persistCredentials: true
```

If this is set, it leaves the OAuth token that is used to fetch the repo, which means the command-line approach that @sima-zhu came up with would work (and we could probably extract the credential from that config file). The security implications of this are probably equivalent to having a variable lying around with this info.

Reply:
Did some investigation, and if we use this option, the config file does indeed contain a GitHub token that could be used.